quick_select/QuickSelect.java

```diff
@@ -0,0 +1,59 @@
+public class QuickSelect {
+    /*
+    * partition function
+    * array        : array on which partitioning has to be done
+    * left      : left index of the partitioning subarray
+    * right     : right index of the partitioning subarray
+    * pivot_idx : pivot index from which partition has to done
+    * store_idx : index of the last element of the left subarray
+    * return    : the index of the last element of the left subarray
+     */
+    public static int partition(int[] array, int left, int right, int pivot_idx) {
+        int pivot_value = array[pivot_idx];
+        int temp = array[right];
+        array[right] = array[pivot_idx];
+        array[pivot_idx] = temp;
+        int store_idx = left;
+        while (left < right){
+            if (array[left] < pivot_value){
+                temp = array[store_idx];
+                array[store_idx] = array[left];
+                array[left] = temp;
+                store_idx++;
+            }
+            left++;
+        }
+        temp = array[right];
+        array[right] = array[store_idx];
+        array[store_idx] = temp;
+        return store_idx;
+    }
+
+    /*
+    * Quick Select function
+    * left        : left index of the subarray
+    * right       : right index of the subarray
+    * pos         : position to find the element using quick sort
+    * pivot_index : pivot index
+    * return      : the value of element at pos place in the sorted array
+     */
+    public static int quickSelect(int[] array, int left, int right, int pos) {
+        int pivot_index;
+        if (left == right)
+            return array[left];
+        pivot_index = right - 1;
```

Comment:
You are considering the last element as pivot I think, but while calling the function you are already passing 0-based index. Please check if it is correct?